components/widgets/inputs/password_input/password_input.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ChangeEventHandler, FocusEventHandler, useState} from 'react';
+import {useIntl} from 'react-intl';
+import classNames from 'classnames';
+
+import Input, {SIZE} from '../input/input';
+
+import './password_input.scss';
+
+type PasswordInputProps = {
+    className?: string;
+    value: string;
+    onChange: ChangeEventHandler<HTMLInputElement>;
+    onBlur?: FocusEventHandler<HTMLInputElement>;
+    onFocus?: FocusEventHandler<HTMLInputElement>;
+    hasError?: boolean;
+    error?: string;
+    createMode?: boolean;
+    disabled?: boolean;
+    inputSize?: SIZE;
+};
+
+const PasswordInput = React.forwardRef(({className, value, onChange, onBlur, onFocus, hasError, error, createMode, disabled, inputSize}: PasswordInputProps, ref?: React.Ref<HTMLInputElement>) => {
```

Comment:
nit: place the variables in new lines so it is easier to read. Wondering why we don't have the maximum characters restriction per line in our linter.